libs/shared/src/test/java/io/crate/common/StringUtilsTest.java

```diff
@@ -44,4 +44,39 @@ public void test_split_empty_str() {
         var parts = StringUtils.splitToList('.', "");
         assertThat(parts).containsExactly("");
     }
+    
+    @Test
+    public void test_tryParseLong() {
+        /* Created after reviewing https://github.com/AdoptOpenJDK/openjdk-jdk/blob/master/test/jdk/java/lang/Long/ParsingTest.java */                 
+        boolean wasLong = false;
+        long[] outputLong = new long[1];
+
+        wasLong = StringUtils.tryParseLong("-123456", outputLong);
+        assertThat(wasLong).isEqualTo(true);
+        assertThat(outputLong[0]).isEqualTo(-123456);
+
+        wasLong = StringUtils.tryParseLong("9223372036854775808", outputLong);
+        assertThat(wasLong).isEqualTo(false);
+        
+        wasLong = StringUtils.tryParseLong("/", outputLong);
+        assertThat(wasLong).isEqualTo(false);
+
+        wasLong = StringUtils.tryParseLong("+", outputLong);
+        assertThat(wasLong).isEqualTo(false);
+
+        wasLong = StringUtils.tryParseLong("", outputLong);
+        assertThat(wasLong).isEqualTo(false);
+
+        wasLong = StringUtils.tryParseLong("++", outputLong);
+        assertThat(wasLong).isEqualTo(false);
+
+        wasLong = StringUtils.tryParseLong("+-", outputLong);
+        assertThat(wasLong).isEqualTo(false);
+
+        wasLong = StringUtils.tryParseLong("--100", outputLong);
+        assertThat(wasLong).isEqualTo(false);
+
+        wasLong = StringUtils.tryParseLong("*100", outputLong);
+        assertThat(wasLong).isEqualTo(false);
```

Comment:
I think we can drop those scenarios - `*100` falls to same code branch as `"/"` - first char is invalid.
and other 2 cases falls to same branch as `"++"` - second char is valid sign (`+` or `-`) but has to be a digit.